src/sentry/issues/issue_velocity.py

```diff
@@ -133,35 +133,39 @@ def calculate_threshold(project: Project) -> Optional[float]:
         tenant_ids={"referrer": REFERRER, "organization_id": project.organization.id},
     )
 
-    result = raw_snql_query(request, referrer=REFERRER)["data"]
-    if len(result) == 0:
-        return None
-
     try:
-        return result[0][THRESHOLD_QUANTILE["name"]]
+        result = raw_snql_query(request, referrer=REFERRER)["data"]
     except Exception:
         logger.exception(
-            "Unexpected shape for threshold query results",
-            extra={"project_id": project.id, "result": json.dumps(result)},
+            "sentry.issues.issue_velocity.calculate_threshold.error",
+            extra={"project_id": project.id},
         )
         return None
 
+    return result[0][THRESHOLD_QUANTILE["name"]]
+
 
-def update_threshold(project: Project, threshold_key: str, stale_date_key: str) -> float:
+def update_threshold(
+    project: Project,
+    threshold_key: str,
+    stale_date_key: str,
+    stale_threshold: Optional[float] = None,
+) -> float:
     """
     Runs the calculation for the threshold and saves it and the date it is last updated to Redis.
-    If the threshold is NaN, we save it as 0 at the normal TTL. If the threshold is None (due to errors
-    in the query or any other reason), we save it as 0 at the shortened TTL.
+    If the threshold is NaN, we save it as 0 at the normal TTL. If the threshold is None (due to
+    Snuba errors), we save it as the stale threshold if we have one, otherwise 0, with the fallback
+    TTL.
     """
     threshold = calculate_threshold(project)
     ttl = DEFAULT_TTL
     if threshold is None:
-        logger.error(
-            "Velocity threshold couldn't be calculated, query returned nothing",
-            extra={"project_id": project.id},
+        logger.warning(
+            "sentry.issues.issue_velocity.update_threshold.fallback",
+            extra={"project_id": project.id, "fallback_value": stale_threshold},
         )
-        threshold = 0
-        ttl = NONE_TTL
+        threshold = stale_threshold if stale_threshold is not None else 0
```

Comment:
this needs to be refactored, i'm realizing the stale threshold can't be accessed after the fallback ttl expires and in the case snuba is down for more than 10 minutes this isn't ideal